pkg/ingester/ingester.go

```diff
@@ -937,9 +1055,9 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 		// TODO(jesus.vazquez) If we had too many old samples we might want to
 		// extend the fast path to fail early.
 		if outOfOrderWindow <= 0 && minAppendTimeAvailable &&
-			len(ts.Samples) > 0 && len(ts.Exemplars) == 0 && allOutOfBounds(ts.Samples, minAppendTime) {
-			stats.failedSamplesCount += len(ts.Samples)
-			stats.sampleOutOfBoundsCount += len(ts.Samples)
+			len(ts.Samples) > 0 && len(ts.Histograms) == 0 && len(ts.Exemplars) == 0 && allOutOfBounds(ts.Samples, minAppendTime) {
```

Comment:
note to reviewers: I'm particularly interested on feedback about this @codesome @jesusvazquez

Reply:
I dont have a lot of context but my read from this is normal samples and histograms are separated in the timeseries context. 

If that is the case then this is okey.

The question that comes to mind is what happens with the stats below that report the amount of samples dropped as well as the out of bounds. Shouldnt we report back the histograms dropped as well? Do we need a specific metric for that or should we report the addition of samples + histograms?

Reply:
Metrics are updated to have a new label called type, as in  `{type="float"}`  or `{type="histogram"}`.

Reply:
If a timeseries has only one sample that is a histogram is it accounted for inside ts.Samples? or only inside ts.Histograms

Reply:
They have separate loops, so counted for ts.Histograms: https://github.com/grafana/mimir/blob/112fb2c76bcc21ec730c7d45795cce72e3bb4fee/pkg/ingester/ingester.go#L1078

I thought about merging the two lists on time, but every such conversation ends up with the same argument: normally a series contains one kind of data, so that's what we need to fully support and optimize for.

Reply:
Right I understand now.

I'm thinking maybe we want to change it to `&& len(ts.Histograms) >= 0` and then report histograms out-of-bounds. After all there is no out-of-order support for them yet.

Reply:
len(x) >= 0 is always true, so basically we'd make a decision based on samples alone. So we might be throwing away some histograms.

Reply:
We should be treating float and histogram samples in the same way. They both are still a sample of a timeseries; only the data type happens to be different. This current change might unnecessarily drop some histograms, we don't want to do that. I would suggest we do this fast path for both float and histograms. We can "split" it into two parts where we loop ts.Samples if not all out of bound and similarly for histograms, without clubbing the fast path.

Reply:
I've updated the fast path to take into account histograms as well and added two basic tests in ingester_test.go

Reply:
continued discussion in Marco's comment